### PR TITLE
add filter department user consumer

### PR DESCRIPTION
**Pull request by borovHokage**

### CODE REVIEW

File: src/DepartmentService.Broker/FilterDepartmentsUsersConsumer.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+    public async Task<List<DepartmentFilteredData>> GetDepartmentFilteredData(IFilterDepartmentsRequest request)
+    {
+      List<DbDepartment> dbPosition = await _repository.GetDepartmenDataAsync(request.DepartmentsIds);
+
+      return dbPosition.Select(
+        pd => new DepartmentFilteredData(
+          pd.Id,
+          pd.Name,
+          pd.Users.Where(u => u.IsActive).Select(u => u.UserId).ToList())).ToList();
```
Comment (MariyaSin) [inline src/DepartmentService.Broker/FilterDepartmentsUsersConsumer.cs:26]: no need select IsActiveUsers, you done it in repo

File: src/DepartmentService.Data/DepartmentRepository.cs [hunk 2]
```diff
@@ -82,6 +82,12 @@ await dbDepartments
           .ToListAsync(),
         await dbDepartments.CountAsync());
     }
+    public async Task<List<DbDepartment>> GetDepartmenDataAsync(List<Guid> departmentIds)
```
Comment (MariyaSin) [inline src/DepartmentService.Data/DepartmentRepository.cs:85]: rename

File: src/DepartmentService.Data/DepartmentUserRepository.cs [hunk 3]
```diff
@@ -65,7 +65,7 @@ public async Task<DbDepartmentUser> GetAsync(Guid userId, bool includeDepartment
         .FirstOrDefaultAsync(u => u.IsActive && u.UserId == userId);
     }
 
-    public async Task<bool> ChangeDirectorAsync(Guid departmentId, Guid newDirectorId)
+      public async Task<bool> ChangeDirectorAsync(Guid departmentId, Guid newDirectorId)
```
Comment (MariyaSin) [inline src/DepartmentService.Data/DepartmentUserRepository.cs:68]: fix tab

File: src/DepartmentService.Broker/FilterDepartmentsUsersConsumer.cs (hunk 1 above)
Comment (borovHokage) [reply src/DepartmentService.Broker/FilterDepartmentsUsersConsumer.cs:26]: fixed

File: src/DepartmentService.Data/DepartmentUserRepository.cs (hunk 3 above)
Comment (borovHokage) [reply src/DepartmentService.Data/DepartmentUserRepository.cs:68]: fixed

File: src/DepartmentService.Data/DepartmentRepository.cs (hunk 2 above)
Comment (borovHokage) [reply src/DepartmentService.Data/DepartmentRepository.cs:85]: fixed

File: src/DepartmentService.Broker/FilterDepartmentsUsersConsumer.cs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+      _repository = repository;
+    }
+
+    public async Task Consume(ConsumeContext<IFilterDepartmentsRequest> context)
+    {
+      List<DepartmentFilteredData> departmentFilteredData = await GetDepartmentFilteredData(context.Message);
+
+      await context.RespondAsync<IOperationResult<IFilterDepartmentsResponse>>(
+        OperationResultWrapper.CreateResponse((_) => IFilterDepartmentsResponse.CreateObj(departmentFilteredData), context));
+
```
Comment (MariyaSin) [inline src/DepartmentService.Broker/FilterDepartmentsUsersConsumer.cs:42]: remove empty line

File: src/DepartmentService.Data/DepartmentRepository.cs [hunk 5]
```diff
@@ -82,6 +82,12 @@ await dbDepartments
           .ToListAsync(),
         await dbDepartments.CountAsync());
     }
+    public async Task<List<DbDepartment>> GetAsync(List<Guid> departmentIds)
```
Comment (MariyaSin) [inline src/DepartmentService.Data/DepartmentRepository.cs:85]: departmentsIds

File: src/DepartmentService.Data/Interfaces/IDepartmentRepository.cs [hunk 6]
```diff
@@ -14,6 +14,8 @@ public interface IDepartmentRepository
   {
     Task<Guid?> CreateAsync(DbDepartment dbDepartment);
 
+    Task<List<DbDepartment>> GetAsync(List<Guid> departmentIds);
```
Comment (MariyaSin) [inline src/DepartmentService.Data/Interfaces/IDepartmentRepository.cs:17]: departmentsIds

**Review by MariyaSin — APPROVED** [review]

File: src/DepartmentService.Broker/FilterDepartmentsUsersConsumer.cs (hunk 4 above)
Comment (borovHokage) [reply src/DepartmentService.Broker/FilterDepartmentsUsersConsumer.cs]: done

File: src/DepartmentService.Data/Interfaces/IDepartmentRepository.cs (hunk 6 above)
Comment (borovHokage) [reply src/DepartmentService.Data/Interfaces/IDepartmentRepository.cs]: done

**Review by ndoriomedov — APPROVED** [review]

**Pull request merged**